### PR TITLE
Document IPC for JSON-RPC

### LINKED ISSUE #1025: Document IPC transport for JSON-RPC APIs
Document how to use the socket interface for JSON-RPC, and any reference content. See https://github.com/hyperledger/besu/pull/3695.

### PR DESCRIPTION
**Pull request by rolandtyler**
Signed-off-by: Roland Tyler <roland.tyler@consensys.net>

## Pull request checklist

Use the following list to make sure your PR fits the Besu documentation quality standard.

### Before creating the pull request

Make sure that:

- [x] [All commits in this PR are signed off for the DCO](https://wiki.hyperledger.org/display/BESU/DCO).
- [x] You've read the [contribution guidelines](https://wiki.hyperledger.org/display/BESU/Documentation).
- [x] You've [previewed your changes locally](https://wiki.hyperledger.org/display/BESU/Preview+the+documentation).

## Describe the change
Document IPC transport for JSON-RPC

## Issue fixed
Fixes #1025 

## Impacted parts

For content changes:

- [x] Documentation content
- [ ] Documentation page organization

For tool changes:

- [ ] CircleCI workflow
- [ ] Build and QA tools (for example, lint or vale)
- [ ] MkDocs templates
- [ ] MkDocs configuration
- [ ] Python dependencies
- [ ] Node dependencies and JavaScript
- [ ] Read the Docs configuration
- [ ] GitHub integration

## After creating your PR and tests have finished

Make sure that:

- [x] You've fixed any issues raised by the tests.
- [x] You've [previewed your changes on Read the Docs](https://wiki.hyperledger.org/display/BESU/Preview+the+documentation)
  and added a [preview link](#preview).

### CODE REVIEW

File: docs/HowTo/Interact/APIs/Using-JSON-RPC-API.md [hunk 1]
```diff
@@ -2,12 +2,19 @@
...
 ---
 
-# JSON-RPC over HTTP and WebSockets
+# JSON-RPC over HTTP, WebSockets and IPC
 
 To enable JSON-RPC over HTTP or WebSockets, use the
 [`--rpc-http-enabled`](../../../Reference/CLI/CLI-Syntax.md#rpc-http-enabled) and
 [`--rpc-ws-enabled`](../../../Reference/CLI/CLI-Syntax.md#rpc-ws-enabled) options.
 
+To enable JSON-RPC over an IPC socket, use the
```
Comment (bgravenorst) [inline docs/HowTo/Interact/APIs/Using-JSON-RPC-API.md:11]: We should perhaps mention the default path here? or link to the "Socket path" topic?

Comment (rolandtyler) [reply docs/HowTo/Interact/APIs/Using-JSON-RPC-API.md]: Added link

File: docs/HowTo/Interact/APIs/API.md [hunk 2]
```diff
@@ -50,6 +55,15 @@ The default ports are:
 
 Ports must be [exposed appropriately](../../Find-and-Connect/Managing-Peers.md#port-configuration).
 
+## Socket path
+
+To specify the socket path for the IPC socket, use the `--Xrpc-ipc-path` option.
+The default path is `besu.ipc`.
```
Comment (macfarla) [inline docs/HowTo/Interact/APIs/API.md:61]: in the Besu data directory

File: docs/HowTo/Interact/APIs/API.md [hunk 3]
```diff
@@ -16,8 +16,13 @@ The following sections provide information about JSON-RPC, RPC Pub/Sub, and Grap
...
 [`--rpc-http-enabled`](../../../Reference/CLI/CLI-Syntax.md#rpc-http-enabled),
-[`--ws-http-enabled`](../../../Reference/CLI/CLI-Syntax.md#rpc-ws-enabled), and
-[`--graphql-http-enabled`](../../../Reference/CLI/CLI-Syntax.md#graphql-http-enabled) options.
+[`--ws-http-enabled`](../../../Reference/CLI/CLI-Syntax.md#rpc-ws-enabled),
+[`--graphql-http-enabled`](../../../Reference/CLI/CLI-Syntax.md#graphql-http-enabled), and
+`--Xrpc-ipc-enabled` options.
+
+!!! caution
+
+    `--Xrpc-ipc-enabled` is an experimental option.
```
Comment (macfarla) [inline docs/HowTo/Interact/APIs/API.md:25]: we don't normally doc experimental options - have we deliberately decided to do so this time?

Comment (rolandtyler) [reply docs/HowTo/Interact/APIs/API.md:25]: @macfarla, we typically don't, but I followed the precedent for other experimental features and included them here (and not the main CLI reference page) in order to document using IPC. Should I hold off until they aren't considered experimental?

**Review by NicolasMassart — APPROVED** [review]

File: docs/HowTo/Interact/APIs/API.md (hunk 3 above)
Comment (macfarla) [reply docs/HowTo/Interact/APIs/API.md:25]: that's fine as long as we considered it

**Review by macfarla — APPROVED** [review]

File: docs/HowTo/Interact/APIs/API.md [hunk 4]
```diff
@@ -6,7 +6,7 @@ description: Hyperledger Besu API
 
 Access the [Hyperledger Besu API](../../../Reference/API-Methods.md) using:
 
-* [JSON-RPC over HTTP or WebSockets](Using-JSON-RPC-API.md)
+* [JSON-RPC over HTTP, WebSockets or IPC](Using-JSON-RPC-API.md)
```
Comment (alexandratran) [inline docs/HowTo/Interact/APIs/API.md:9]:
```suggestion
* [JSON-RPC over HTTP, WebSocket, or IPC](Using-JSON-RPC-API.md)
```

**Review by alexandratran — APPROVED** [review]

**Pull request merged**